### PR TITLE
Release 0.19.0

### LINKED ISSUE #1303: Quantity warning in "Edit Variant" is not working correctly
### Expected behavior

Quantity warning should display an accurate warning to the user that the quantity of an item might not be available.
### Actual Behavior

Quantity warning is not correctly displaying.
screen shot 2016-08-16 at 5 44 35 pm
### Steps to Reproduce the Behavior

Add quantity warning when editing a product.
Change inventory to less than that number

### LINKED ISSUE #1412: use email package hasPermissions
Related to #1367. Update `Roles` in the core email methods to use Reaction.hasPermissions, and use the `email` package permissions.

### LINKED ISSUE #1615: Flickering images when mousing over secondary images
### Expected behavior
Whatever image you mouse-over should be shown in the "hero" spot.

### Actual Behavior
Images flicker in and out of hero spot.

### Steps to Reproduce the Behavior
1. Add more than one image to a variant or option with irregular sizes (one tall and one wide)
1. Mouseover the secondary image
1. Observe that when the image is swapped in, your mouse is no longer over the image so then the image is swapped back, where it is now over the image and so on...

This gif doesn't quite capture it as it happens too fast but you can sort of see it.

mouseover_image

### LINKED ISSUE #1695: IOS / safari flex-wrap not working
The flex-wrap IOS safari issue is pretty well known and unfortunately it seems to be having an impact of the reactioncommerce platform. 

BTW: on a Mac chrome, firefox and safari seem ok and on mobile chrome works as well. The only time I find this error is on an iPhone.

Two example and fixes I would like to share include:

Example 1 - 
Viewing the product details like in the screenshot below, the page is trying to render 2 columns which isn't wrapping. The easiest way I found to fix this is to force the flex-basis to 100% either with .flex(1 1 100%) or -webkit-flex-basis: 100%.

Solution:

.rui.items.flex .item.half {
  flex: 1 1 50%;
  @media @mobile {
    -webkit-flex-basis: 100%;
  }
}
screen shot 2017-01-07 at 11 27 59 am

Example 2 - 
Product grid...
Solution:

  // Small product: default
  &.product-small {
    @media screen and (max-width: @screen-xs-max) {
      -webkit-flex-basis: 100%;
    }
  }

screen shot 2017-01-07 at 11 38 41 am

### LINKED ISSUE #1708: Some i18n mapping fixes
Two that I have found so far:

1. In "notificationDropdown.js" it refers to "notifications.body.noNotifcations" but should be "notifications.body.**noNotifications**"

2. "availableOptions" should be under "productDetail" and not shop

### LINKED ISSUE #1709: Allow users to change their default currency across the app using a dropdown
We will introduce a currency change dropdown, similar to the one for language - practically a copy, which is now ready.
The dropdown will make use of the existing functionality, which is solely based on the user's Locale.

The value set from the dropdown will be saved in `LocalStorage`, as well as in `users[n].profile.currency`. Just like language is.
This will have the highest precedence and will be checked before the current locale functionality.

With that in place, it should be easy to put that dropdown in the header so that each user can choose their favourite currency.

Any thoughts, let us know.

### LINKED ISSUE #1730: Archived products still show up in Product Search
### Expected behavior
Once a product is archived, it should no longer appear in product search

### Actual Behavior
It does.

### Steps to Reproduce the Behavior
1. Add and publish a product.
1. Verify it shows up in product search
1. Archive product and publish the change
1. Search for the archived product
1. Observe it still shows up in results

### LINKED ISSUE #1741: Add tracking number to Shipment email

### LINKED ISSUE #1746: Sidebar behavior fixes
Improve the user experience of the sidebar (ActionView)

- Action and detail view
- Collapsable panels everywhere!

### LINKED ISSUE #1747: Grid Page UI behavior updates
Update behavior on grid page: 
1. 1 click selects
2. Double click takes you into the product
3. Click again does nothing once selected
4. Click on the BG deselects everything
5. Add indicator for state shift (for now let's use a new color on the selector stroke to indicate when it's been edited) 
6. Auto select cloned products.

### LINKED ISSUE #1749: Save Admin / Customer preview state to the user profile
We should save the current view state - admin or customer preview - to a users meteor profile, so we can keep the state saved across the app, and rid the app of unneeded queryparams (`as=customer`).

### LINKED ISSUE #1756: Card Behavior Tweaks
- Add disclosure arrow to collapsable cards
- Add ability to add switch components to card headers (useful for enabling settings)
- Cleanup card borders when used inside of ActionView

### PR DESCRIPTION
**Pull request by mikemurray**
# Major Features

- Reorganized administration views. Dashboard now opens in the side panel. Toolbar moved to top.
- Reorganized shipping dashboard views
- Avalara Tax Reporting (#1704)
- Shippo Plugin (PR #1618) @prosf @lcampanis

## Other Features

- All packages in `imports/plugins/` now have a local `server/i18n` directory for translations.
- Improvements to the overall and feel of the default theme.
- React MultiSelect component implementation ( PR #1681 ) @lcampanis
- React no-ui Slider component implementation ( PR #1680 ) @lcampanis

### 0.19.0

- Fix CircleCI unset var issue with production deployment ( PR #1724  )
- Fix Product Not Found infinite spinner ( PR #1725 )
- IOS / safari flex-wrap not working ( PR #1727 / Resolves #1695 ) @chluke123
- Allow users to change their default currency across the app using a dropdown ( PR #1728 / Resolves #1709 ) @lcampanis 
- Shippo tracking status ( PR #1735 ) @lcampanis 
- Fix inventory notifications ( PR #1736 / Resolves #1303 )
- Update Product Publications Data ( PR #1738 )
- Update dashboard and settings related registry entries  ( PR #1739 )
- Silence of the Logs ( PR #1745 )
- Update viewAs [Preview] to used profile saved data ( PR #1748 / Resolves #1749 )
- Color fixes ( PR #1750 )
- Grid View Behavior Updates ( PR #1752 / Resolves #1747)
- Fix bodyClass for Not Found page (PR #1754 ) @lcampanis
- Dashboard cleanup (PR #1755 / Resolves #1746 )
- Card, Button UI Updates ( PR #1757 / Resolves #1756 )
- Simplified the new MultiSelect component ( PR #1760 )
- Add tracking / shipping to emails ( PR #1761 / Resolves #1741 )
- Fix Notifications i18n typo ( PR #1765 / Resolves #1708 )
- Fix "sold out" showing on all products with tags ( PR #1776 / Resolves #1303  )
- Prevent archived products from showing in search ( PR #1821 / Resolves #1730 )
- Fix media gallery flickering and missing thumbnail ( PR #1794 / Resolves #1615 #1649 )
- Switch email methods to use Reaction.hasPermission API ( PR #1829 / Resolves #1412 )
- Resolve access-denied error on non-admin checkout ( PR #1833 )

## Contributors

@reactioncommerce/core
@lcampanis
@prosf
@chluke123
@f3rnando

### CODE REVIEW

File: imports/plugins/included/product-variant/client/templates/products/productGrid/productGrid.js [hunk 1]
```diff
@@ -13,8 +13,8 @@ import Sortable from "sortablejs";
 Template.productGrid.onCreated(function () {
   const profile = Meteor.user().profile;
 
-  if (profile && profile.preferences && profile.preferences['reaction-product-variant'] && profile.preferences['reaction-product-variant'].selectedGridItems) {
-    let selectedProducts = profile.preferences['reaction-product-variant'].selectedGridItems;
+  if (profile && profile.preferences && profile.preferences["reaction-product-variant"] && profile.preferences["reaction-product-variant"].selectedGridItems) {
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/client/templates/products/productGrid/productGrid.js]: line too long?

File: imports/plugins/included/shippo/lib/collections/schemas/shippo.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { SimpleSchema } from "meteor/aldeed:simple-schema";
+import { PackageConfig } from "/lib/collections/schemas/registry";
+
+export const ShippoPackageConfig = new SimpleSchema([
+  PackageConfig, {
+    "settings.apiKey": {
+      type: String,
+      label: "API Key",
+      min: 10,
+      optional: true
```
Comment (zenweasel) [inline imports/plugins/included/shippo/lib/collections/schemas/shippo.js]: Should this not be optional?

File: imports/plugins/included/shippo/server/methods/shippoapi.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
+/* eslint camelcase: 0 */
+import Shippo from "shippo";
+import { Meteor } from "meteor/meteor";
+import { Logger } from "/server/api";
+import { SimpleSchema } from "meteor/aldeed:simple-schema";
```
Comment (zenweasel) [inline imports/plugins/included/shippo/server/methods/shippoapi.js]: Import order

File: lib/api/products.js [hunk 4]
```diff
@@ -3,9 +3,10 @@ import { Meteor } from "meteor/meteor";
 import { ReactiveDict } from "meteor/reactive-dict";
 import { FlowRouter } from "meteor/kadira:flow-router-ssr";
 import { getCurrentTag, getShopName } from "/lib/api";
-import { Products } from "/lib/collections";
+import { Products, Revisions } from "/lib/collections";
 import Catalog from "./catalog";
 import { MetaData } from "/lib/api/router/metadata";
+import orderBy from "lodash/orderBy";
```
Comment (zenweasel) [inline lib/api/products.js]: Import order

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "reaction",
   "description": "Reaction is a modern reactive, real-time event driven ecommerce platform.",
-  "version": "0.18.1",
+  "version": "0.20.0",
```
Comment (zenweasel) [inline package.json]: Whoops

File: server/publications/collections/shipping.js [hunk 6]
```diff
@@ -1,16 +1,32 @@
+import { Meteor } from "meteor/meteor";
+import { Match, check } from "meteor/check";
 import { Shipping } from "/lib/collections";
 import { Reaction } from "/server/api";
-
+import { Counts } from "meteor/tmeasday:publish-counts";
```
Comment (zenweasel) [inline server/publications/collections/shipping.js]: Import order

File: imports/plugins/core/ui/client/components/list/list.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { Component, PropTypes } from "react";
```
Comment (mikemurray) [inline imports/plugins/core/ui/client/components/list/list.js]: React is used at runtime when the JSX is transpiled to JS. ( you will receive a runtime error is React is not imported )

File: imports/plugins/included/shippo/lib/collections/schemas/shippo.js (hunk 2 above)
Comment (mikemurray) [inline imports/plugins/included/shippo/lib/collections/schemas/shippo.js]: Leaving it blank means it's not fully configured, and shippo will effectively be disabled. When an API key is entered then it's (really) enabled.

**Comment by aaronjudd** [comment]
It is true that we should still be deprecating use of `Session`, it's also a global that we probably don't want to rely on (without an import).  An issue to deprecate `Session` along with a general issue to resolve the remaining lint items,  should be sufficient to include in the next few releases.

Can we make notes about the upgrade path, and the breaking changes (registry resets might be needed with this update, for instance)

**Pull request merged**